doc/meta.xml

```diff
@@ -250,6 +250,61 @@ meta.platforms = stdenv.lib.platforms.linux;
      </para>
     </listitem>
    </varlistentry>
+   <varlistentry>
+    <term>
+     <varname>tests</varname>
+    </term>
+    <listitem>
+     <para>
+      An attribute set with as values tests. A test is a derivation, which
+      builds successfully when the test passes, and fails to build otherwise. A
+      derivation that is a test requires some <literal>meta</literal> elements
+      to be defined: <literal>needsVMSupport</literal> (automatically filled-in
+      for NixOS tests) and <literal>timeout</literal>.
+     </para>
+     <para>
+      The NixOS tests are available as <literal>nixosTests</literal> in
+      parameters of derivations. For instance, the OpenSMTPD derivation
+      includes lines similar to:
+<programlisting>
+{ /* ... */, nixosTests }:
+{
+  # ...
+  meta.tests = {
+    basic-functionality-and-dovecot-integration = nixosTests.opensmtpd;
```

Comment:
Not sure if this has been discussed before (can't find it with a quick skim), but is there a reason to make this a set instead of a simple list? I feel like having to give every test a new name only reduces redundancy. If there is additional information (like what exactly is tested in this case), maybe renaming the original test or a comment would be better?

Reply:
I had originally implemented this as a list, and @Profpatsch's comment (https://github.com/NixOS/nixpkgs/pull/44439#issuecomment-410447611) made me notice it could be written as an attrset. At the time, it was a good thing, because tests needed to be `import`-ed, and the additional parenthesis in lists are often a source of errors.

Now that the tests are easily accessible from the fixpoint… I don't have any strong sentiment about it. @Profpatsch maybe has?

Reply:
A list is kinda hard to work with, it can only be mapped over and replaced by an other list. There is no (sane) way to reference single elements in it. So if there should ever be the need to work with one of these tests especially (e.g. for overriding or removing it), you don’t have a good reference to work with (and no, the test’s name attribute is not a good reference).

Reply:
I still don't like the "UX" of attribute sets here very much, but your argument makes sense. Using `inherit` would at least remove the redundancy.

So as a summary I'm not quite happy but have no better idea.